lib/libc/common/source/signal/signal.c

```diff
@@ -0,0 +1,14 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <signal.h>
+
+void (*signal(int signum, void (*handler)(int)))(int)
+{
+	ARG_UNUSED(signum);
+	ARG_UNUSED(handler);
+	errno = EINVAL;
+	return -1;
```

Comment:
Apparently l, errno should not be set here.

https://en.cppreference.com/w/c/program/signal
https://linux.die.net/man/2/signal
```suggestion

	return SIG_ERR;
```

Reply:
Please refer to both the ISO C and POSIX specs which have the function return SIG_ERR as well as setting errno. In the case of ISO C, errno is to be set to 'some positive value'. In the case of POSIX, they only list EINVAL as a possible errno value.

Reply:
Ah - indeed! From some version of the iso C spec 

![Screenshot_20250919-055420.png](https://github.com/user-attachments/assets/e53ca4de-f8b2-45e8-8378-99994d85971a)

cppreference.com failed to include that bit of info

Reply:
Revised suggestion:
```suggestion
	errno = EINVAL;
	return SIG_ERR;
```